03-attribute-operations.qmd

```diff
@@ -361,27 +361,38 @@ In that case an inner join can be used:
 pd.merge(world, coffee_data, on='name_long', how='inner')
 ```
 
+An alternative way to join two (Geo)DataFrames is the aptly called `join` function: 
+
+```{python}
+world.join(coffee_data.set_index('name_long'), on='name_long', how='inner')
+```
+
+Note that in this case, we need to set the index of `coffee_data` to the `name_long` values to avoid error messages.
+
```

Comment:
Same as in another comment, personally I prefer to explain the minimal set of methods in detail, and omit mentioning alternatives that do the same thing. Admittedly this is subjective - we can discuss, and either strategy we choose is fine with me.

Reply:
@michaeldorman my geocompr approach is to: a) explaining the minimal set of methods in detail, and b) in some cases, when useful, adding a text block at the end of section listing some alternatives.

Reply:
> @michaeldorman my geocompr approach is to: a) explaining the minimal set of methods in detail, and b) in some cases, when useful, adding a text block at the end of section listing some alternatives.

Excellent! I'm 100% in favor of this approach. That way the reader is aware that the alternatives are optional material that's not explained in detail, and they can investigate it on their own if interested.

Reply:
"join" is also a very particular case, since people coming from SQL backgrouds will search for and find the join function before they find the merge function (heck, even our section title is "join") so readers would be wondering why we are not using the correspondingly named function

Reply:
Good point... FWIW the base R function for table joins is also called `merge`, so the `pandas` naming didn't seem so unnatural to me. 
What is the practical difference between `merge` and `join` in `pandas` then? The documentation doesn't make it clear as far as I could tell.

Reply:
I don't know if there is much of a difference besides that one is a function of a DataFrame and the other is a method that accepts two DataFrame inputs.

Reply:
> I don't know if there is much of a difference besides that one is a function of a DataFrame and the other is a method that accepts two DataFrame inputs.

They're both methods of a `DataFrame`, but `merge` is also a standalone function that accepts two `DataFrame`s... `pandas` definitely has too many ways to do the same thing IMHO